### PR TITLE
Bug 1563220: switch to using docker client CLI

### PR DESCRIPTION
**Pull request by milescrabill**

### CODE REVIEW

File: process/docker.go [hunk 1]
```diff
@@ -62,60 +48,41 @@ func (c *Command) String() string {
...
-	if r.SystemError != nil {
-		return
-	}
 
-	r.SystemError = c.cli.ContainerStart(c.ctx, c.resp.ID, types.ContainerStartOptions{})
-	if r.SystemError != nil {
+	image := "ubuntu"
+
+	// TODO scary injection potential here
+	cmd := exec.CommandContext(c.ctx, "/usr/bin/docker", append([]string{"run", image}, c.cmd...)...)
```
Comment (petemoore) [inline process/docker.go]: We should probably make the docker executable a configuration setting (e.g. on my mac it is `/usr/local/bin/docker`).

Comment (petemoore) [reply process/docker.go]: Don't we need to pull the docker image before using it?

Comment (petemoore) [reply process/docker.go]: Let's also log any commands we run to the worker log.

Comment (milescrabill) [reply process/docker.go]: We're logging commands now. Images get pulled as part of the `run` command, and are kept on disk for future use (not that we support that).

**Pull request merged**